### PR TITLE
Homework 11 task 1

### PR DESCRIPTION
**Pull request by ivanzuev78**
Not sure, that this task should be done in this way, but I don't have any other ideas.

### CODE REVIEW

File: HomeWork_11/hw1.py [hunk 1]
```diff
@@ -1,35 +1,51 @@
...
+    assert ColorsEnum.RED == "RED"
+    assert SizesEnum.XL == "XL"
+    """
+
+    def __getattr__(cls, item):
+        if f"_{cls.__name__}__keys" in cls.__dict__:
+            if item in cls.__dict__[f"_{cls.__name__}__keys"]:
+                return str(item)
+
+    def __iter__(cls):
```
Comment (Cjkjvfnby) [inline HomeWork_11/hw1.py:43]: Is it required?

File: HomeWork_11/hw1.py [hunk 2]
```diff
@@ -1,35 +1,51 @@
...
+        if f"_{cls.__name__}__keys" in cls.__dict__:
+            return (item for item in cls.__dict__[f"_{cls.__name__}__keys"])
+        return ()
+
+    def __len__(cls):
```
Comment (Cjkjvfnby) [inline HomeWork_11/hw1.py:48]: Is it required?

File: HomeWork_11/hw1.py [hunk 3]
```diff
@@ -1,35 +1,51 @@
...
-assert SizesEnum.XL == "XL"
-"""
+    assert ColorsEnum.RED == "RED"
+    assert SizesEnum.XL == "XL"
+    """
+
+    def __getattr__(cls, item):
+        if f"_{cls.__name__}__keys" in cls.__dict__:
+            if item in cls.__dict__[f"_{cls.__name__}__keys"]:
+                return str(item)
```
Comment (Cjkjvfnby) [inline HomeWork_11/hw1.py:41]: it is a string by definition of the __getattr__.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
This approach is a kind of magic.  You can play with a `__new__` method and make a required class in it.  It's optional.

File: HomeWork_11/hw1.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_11/hw1.py:43]: I thought that enum should have `__iter__` and `__len__`.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**